### ISSUE TITLE
JSAPI scroll documentation is inconsistent

### ISSUE THREAD

**Issue by pushplay** [elastic/elasticsearch-js#467]
In the documenation for client.scroll at https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-scroll the parameter scroll is listed as accepting either a Date or number. The example on the same page uses a stringscroll: '30s'. Either the parameter should be documented as also accepting a string or the example needs to be updated.

If the parameter can be a string like '30s' is that also true of other parameters in other functions that accept Date, Number?

**Comment by spalger** [comment]
That's a good question.

The [elasticsearch source](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/rest/action/search/RestSearchScrollAction.java#L58-L80) indicates that the only valid values for the scroll parameter strings followed by a [time unit](https://www.elastic.co/guide/en/elasticsearch/reference/5.0/common-options.html#time-units). 

Best I can tell by looking at the history this has always been the case, but [before 5.0](https://github.com/elastic/elasticsearch/pull/15948) the units were not required.

This does indeed mean that the docs are incorrect.

**Comment by pushplay** [comment]
So just to clarify so I can get the TypeScript declaration file right: parameters named `timeout`, `masterTimeout`, `scroll`, `interval` and `ttl` *must* be strings like `"30s"` (and presumably `timestamp` is correct with `Date, number`).  Yes?

**Comment by spalger** [comment]
Regarding the second question:

I suspect it is the case for many of the other parameters that are marked with `Date, Number`. These parameters are identified with `"type": "time"` in the [rest api spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/create.json#L48) (which might be helpful in your quest to type the elasticsearch client).

The only exception I know of so far is the `timestamp` parameter of the `create`, `index`, and `update` APIs. This parameter is marked as type `time` in the rest spec, but actually expects a date value.. Going to have to look deeper into that one

**Comment by pushplay** [comment]
Awesome.  Thanks for the quick turnaround on this.

**Comment by spalger** [comment]
Sure thing, thank you for helping update the community type definitions!

**Comment by pushplay** [comment]
This is also in the example code for `client.scan`:

```
client.search({
  index: 'myindex',
  // Set to 30 seconds because we are calling right back
  scroll: '30s',
  search_type: 'scan',
  fields: ['title'],
  q: 'title:test'
}```

It looks like `search_type: 'scan'` should now be `searchType: 'query_then_fetch'` and `fields: ['title']` should be `docvalueFields: ['title']`.

**Comment by spalger** [comment]
This should be fixed with the updated that came along with 12.1
